pkg/virtctl/expose/expose.go

```diff
@@ -201,3 +215,25 @@ func (o *Command) RunE(cmd *cobra.Command, args []string) error {
 	fmt.Printf("Service %s successfully exposed for %s %s\n", serviceName, vmType, vmName)
 	return nil
 }
+
+func podNetworkPorts(vmiSpec *v12.VirtualMachineInstanceSpec) []v1.ServicePort {
+	podNetworkName := ""
+	for _, network := range vmiSpec.Networks {
+		if network.Pod != nil {
+			podNetworkName = network.Name
+			break
+		}
+	}
+	if podNetworkName != "" {
+		for _, device := range vmiSpec.Domain.Devices.Interfaces {
+			if device.Name == podNetworkName {
+				ports := []v1.ServicePort{}
+				for i, port := range device.Ports {
+					ports = append(ports, v1.ServicePort{Name: fmt.Sprintf("port-%d", i+1), Protocol: v1.Protocol(port.Protocol), Port: port.Port})
```

Comment:
maybe "k6t-port-..." ?

Reply:
Not sure. I just copied what the kubelet does. What would be the benefit?

Reply:
prevent conflicts with someone calling "kubectl..." on the pod?

Reply:
Ah I see what you want to get at. So that command creates a service with ports where the ports in the service are named `port-${x}`. That is not related to the port name on the Pod. At the end `kubectl` and `virtctl` just create a final service object. If they happen to have the same service-name then only one can succeed. The same like with two parallel kubectl calls.